src/shared/nt-core/withLoader.react.js

```diff
@@ -0,0 +1,27 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import styles from './withLoader.styl'
+
+function withLoader (Component) {
+  return class extends React.Component {
+    static propTypes = {
+      data: PropTypes.shape({
+        loading: PropTypes.bool,
+        error: PropTypes.bool
+      }).isRequired
+    }
+
+    render () {
+      return (this.props.data.loading) ? (
+        <div className={styles.loader}>
+          <p>data fetching...</p>
+        </div>
+      ) : (
+        <Component {...this.props} />
+      )
+    }
+  }
+}
+
+export { withLoader }
```

Comment:
**update**
- I have to change the export style to be an object because I created another higher order component. 
- This component file should be called `withLoader` because `withLoader` is the main responsibility, `handleError` component is a subcomponent to do a few responsibility.